composer/metrics/nlp.py

```diff
@@ -431,6 +432,7 @@ def __init__(self, dist_sync_on_step: bool = False):
         self.add_state('total', default=torch.tensor(0.0), dist_reduce_fx='sum')
 
     def update(self, batch: dict, output_logits: torch.Tensor, labels: torch.Tensor):
+        output_logits = torch.softmax(output_logits, dim=2)
```

Comment:
I think HF forward is going to provide unnormalized logits so I need to use softmax in multiple choice in order to ensure we get real perplexities. The unnormalized logits of different MC answers won't be directly to comparable to one another b/c they could be scaled differently. I'm not 100% certain of the above logic, does it make sense to you?

Reply:
i'm a bit confused because this line is in `InContextLearningMultipleChoiceAccuracy`, not the ECE class, so this doesn't have any impact on the ECE, right?

Reply:
oh, are you saying the old MC metric was actually wrong?

Reply:
i believe that `F.cross_entropy` handles the normalization for you (https://pytorch.org/docs/stable/generated/torch.nn.functional.cross_entropy.html#torch.nn.functional.cross_entropy) "input ([Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)) – Predicted unnormalized logits; see Shape section below for supported shapes."

Reply:
yeah I think you're right! good catch